src/user.js

```diff
@@ -93,6 +98,19 @@ class User {
     }
     return new User();
   }
+
+  static getUser(clientId, handler) {
+    // when login is made via github, `patternMatch` will have an extra index entry with the user's GH username
+    // e.g., ['mozilla-auth0/github|0000/helfi92, 'mozilla-auth0/github|0000', 'helfi92']
+    const patternMatch = CLIENT_ID_PATTERN.exec(clientId);
+    const encodedUserId = patternMatch[1].replace(User.identityPrefix(patternMatch[1]), '');
+
+    const userPromise = handler.userFromIdentity(
+      decodeURIComponent(encodedUserId)
+    );
```

Comment:
Let's move all of this into the handler itself -- so the handler would have a `userFromClientId` method.  Then it can do any decoding that it wants.  This will also allow the handler to generate an identityId of `github|1234/djmitche` but only look at the `github|1234` when looking up the user in the management API.